backend/shared/src/distribute-kyc-bonus.ts

```diff
@@ -0,0 +1,48 @@
+import { SupabaseDirectClient } from 'shared/supabase/init'
+import { runTxnFromBank } from 'shared/txn/run-txn'
+
+export async function distributeKycBonus(
+  pg: SupabaseDirectClient,
+  userId: string
+) {
+  return await pg.tx(async (tx) => {
+    const data = await tx.oneOrNone<{
+      reward_amount: number
+      claimed: boolean
+    }>(
+      'select reward_amount, claimed from kyc_bonus_rewards where user_id = $1',
+      [userId]
+    )
+
+    if (data?.claimed) {
+      return
+    }
+
+    const reward = Math.max(data?.reward_amount ?? 0, MIN_KYC_REWARD)
```

Comment:
MIN_KYC_REWARD should be SWEEPS_INITIAL_BONUS from economy.ts instead

Fetching from kyc_bonus_rewards is the special case; not the other way around